src/gui/topbar_subwindows/military_subwindows/gui_build_unit_large_window.hpp

```diff
@@ -46,6 +46,13 @@ class build_unit_close_button : public button_element_base {
 
 		state.ui_state.build_unit_window->set_visible(state, false);
 	}
+	void on_update(sys::state& state) noexcept override {
+		if(state.world.nation_get_disarmed_until(state.local_player_nation)) {
```

Comment:
the until disarmed date doesn't go away when being disarmed ends. You are disarmed if there is a valid date set there *and* the current date is prior to that day (see the guts of the build unit command)